sa/model.go

```diff
@@ -726,7 +726,8 @@ func newAuthzReqToModel(authz *sapb.NewAuthzRequest) (*authzModel, error) {
 // authzModel storage representation.
 func authzPBToModel(authz *corepb.Authorization) (*authzModel, error) {
 	am := &authzModel{
-		IdentifierValue: authz.Identifier,
+		IdentifierType:  identifierTypeToUint[string(identifier.DNS)],
```

Comment:
This change being present here is surprising! If we only intend to use this function for DNS identifiers we should probably comment as much and consider a different name for the function.

Reply:
I definitely don't intend for this function to only be used for DNS identifiers.

The issue here is that the input corepb.Authorization doesn't have any indicator of the identifier type. It just comes in with an "Identifier" (now renamed to "DnsName", because that's what it is), and no type information. Previously, this function was leaving the IdentifierType field of the output authzModel _empty_. I think that's terrible -- it's just happenstance that the zero value of IdentifierType is also the integer we use to indicate the "dns" identifier type in the database.

So the only change here is making the hardcoding of "IdentifierType: 0" explicit instead of implicitly relying on the zero value. And then when the input corepb.Authorization starts carrying type info, we'll update this function to pass it through instead of hardcoding it.